### PR TITLE
Unicorn JS: Upgrade from Babel 5 to 6

### PR DESCRIPTION
**Pull request by brev**
Unicorn JS: Upgrade from Babel 5 to 6

- Make changes to get Babel working again
  - Get past decorator problems with `babel-plugin-transform-decorators-legacy`
  - Get Babel5-style export defaults with `babel-plugin-add-module-exports`
    - Electron .remote() Client modules have to use new Babel6-style `.default` import
- Upgrade most packages
- Move Logger config out to own file
- Match header license boilerplate style to other pages
- etc.

@lscheinkman @marionleborgne ready for review
cc @cbaranski

### CODE REVIEW

File: unicorn/.babelrc [hunk 1]
```diff
@@ -1,3 +1,4 @@
 {
-  "optional": [ "es7.classProperties" ]
+  "plugins": ["add-module-exports", "transform-decorators-legacy"],
+  "presets": ["es2015", "stage-0", "react"]
```
Comment (lscheinkman) [inline unicorn/.babelrc]: Are you sure about `stage-0`? 
Gulp was using `stage-1`. See https://github.com/numenta/numenta-apps/pull/520/files#diff-7c92219528c7020722542f623e43b797L70

**Comment by lscheinkman** [comment]
@brev I am getting the following exception:
```
Uncaught TypeError: DatabaseClient is not a function(anonymous function) 
@ app.js:59__webpack_require__ 
@ bootstrap f68038b5294c5cf92b88:19(anonymous function) 
@ bootstrap f68038b5294c5cf92b88:39__webpack_require__ 
@ bootstrap f68038b5294c5cf92b88:19(anonymous function) 
@ bootstrap f68038b5294c5cf92b88:39(anonymous function) 
@ bootstrap f68038b5294c5cf92b88:39
/deep/ combinator is deprecated. See https://www.chromestatus.com/features/6750456638341120 for more details.
```

File: unicorn/.babelrc (hunk 1 above)
Comment (brev) [inline unicorn/.babelrc]: Could use any we want, I just went for everything. I could try 1 again if you prefer?

**Comment by brev** [comment]
@lscheinkman When do you see that error, right away?  Or is it triggered by a UI action?  I cannot reproduce yet, but will keep trying.

**Comment by lscheinkman** [comment]
Here are the steps:
```
npm run clean
npm install
npm run desktop
```
The application does not start and the exception is shown on the console.

File: unicorn/.babelrc (hunk 1 above)
Comment (lscheinkman) [inline unicorn/.babelrc]: everything sounds good

**Comment by brev** [comment]
@lscheinkman still cannot reproduce, I'll try from my home machine tonight. @marionleborgne do you have a chance to test quickly? thanks

**Comment by brev** [comment]
@lscheinkman do you happen to have any `babel*` npm packages installed globally? Trying to figure out where a conflict might be

**Comment by lscheinkman** [comment]
No, all my `node` modules are installed locally.

**Comment by brev** [comment]
@lscheinkman got it! A clean git clone does the trick, I see the error, working on fixing now. thanks.

**Comment by brev** [comment]
OK @lscheinkman please try again, thanks

**Comment by lscheinkman** [comment]
:+1: It works now.
Thank you for updating.

**Pull request merged**